tracer/build/_build/CodeGenerators/CallTargetsGenerator.cs

```diff
@@ -0,0 +1,743 @@
+                        case nameof(InstrumentAttributeProperties.MethodName):
+                            methodName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.ReturnTypeName):
+                            returnTypeName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.MinimumVersion):
+                            minimumVersion = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.MaximumVersion):
+                            maximumVersion = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.ParameterTypeNames):
+                            parameterTypeNames = GetStringArray(namedArgument.Argument.Value);
+                            break;
+                        case nameof(InstrumentAttributeProperties.CallTargetType):
+                            //callTargetType = (namedArgument.Argument.Value as INamedTypeSymbol)?.ToDisplayString();
+                            break;
+                        case nameof(InstrumentAttributeProperties.CallTargetIntegrationKind):
+                            integrationKind = namedArgument.Argument.Value as int?;
+                            break;
+                        case nameof(InstrumentAttributeProperties.InstrumentationCategory):
+                            instrumentationCategory = (InstrumentationCategory)(namedArgument.Argument.Value as uint?).GetValueOrDefault();
+                            break;
+                        default:
+                            hasMisconfiguredInput = true;
+                            break;
+                    }
+
+                    if (hasMisconfiguredInput)
+                    {
+                        break;
+                    }
+                }
+
+                (ushort Major, ushort Minor, ushort Patch) minVersion = default;
+                (ushort Major, ushort Minor, ushort Patch) maxVersion = default;
+                TryGetVersion(minimumVersion, ushort.MinValue, out minVersion);
+                TryGetVersion(maximumVersion, ushort.MaxValue, out maxVersion);
+
+                foreach (var assembly in assemblyNames ?? new[] { assemblyName })
+                {
+                    foreach (var t in typeNames ?? new[] { typeName })
+                    {
+                        res.Add(
+                            new CallTargetDefinitionSource(
+                                integrationName: integrationName!,
+                                assemblyName: assembly!,
+                                targetTypeName: t!,
+                                targetMethodName: methodName!,
+                                targetReturnType: returnTypeName!,
```

Comment:
nit: I think we should add some validation that these values _have_ been set correctly and aren't null or empty (and just throw if any of them are)

Reply:
Done